rskj-core/src/main/java/co/rsk/util/EthSwapUtil.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2024 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.util;
+
+import co.rsk.core.Coin;
+import co.rsk.core.RskAddress;
+import co.rsk.core.bc.ClaimTransactionInfoHolder;
+import co.rsk.db.RepositorySnapshot;
+import org.ethereum.config.Constants;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.core.CallTransaction;
+import org.ethereum.core.SignatureCache;
+import org.ethereum.core.Transaction;
+import org.ethereum.crypto.HashUtil;
+import org.ethereum.solidity.SolidityType;
+import org.ethereum.util.ByteUtil;
+import org.ethereum.vm.DataWord;
+import org.ethereum.vm.GasCost;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.List;
+
+public class EthSwapUtil {
+    private static final String SWAPS_MAP_POSITION = "0000000000000000000000000000000000000000000000000000000000000000";
+
+    private EthSwapUtil() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    public static byte[] calculateSwapHash(Transaction newTx, SignatureCache signatureCache) {
```

Comment:
This method is being used only once inside this class, is it needed to be public?

Reply:
Not really, at least not right now. My reasoning was to make it public in cases we need it somewhere out side the class in the future but for now I've changed it to private.